httpx/_models.py

```diff
@@ -912,19 +913,28 @@ def read(self) -> bytes:
             self._content = b"".join(self.iter_bytes())
         return self._content
 
-    def iter_bytes(self) -> typing.Iterator[bytes]:
+    def iter_bytes(self, chunk_size: int = None) -> typing.Iterator[bytes]:
```

Comment:
Are you sure about chunk_size with default `None`?

 I do agree that it looks more suitable, but `requests` provides us with defaults [chunk_size=1](https://2.python-requests.org/en/master/_modules/requests/models/#Response.iter_content) or [512](https://2.python-requests.org/en/master/_modules/requests/models/#Response.iter_lines)

Reply:
In this PR, when `chunk_size=None` we just return the input `content` unchanged, as one single big chunk.

Yes, this would deviate from what Requests seems to do, but:

- Setting a non-`None` default would break backward compatibility on our side.
- Defaulting to "transparently pass the chunk sent by the server" is probably the most reasonable approach _anyway_.

That said, we'd need to add this deviation from Requests to the compatibility guide. :+1: